drivers:platform:maxim:max32670 Initial implementation of IRQ platform drivers

mphalke opened the pull request:
The IRQ platform drivers for MAX32670 are same as that of MAX32660.
The source files are copied from MAX32660 folder and below changes made:
1. Renamed 'max32660.h' include header with 'max32670.h'.
Rest of the code is unchanged. The code is tested on MAX32670 EVKIT for generating GPIO based IRQ event.
Original author information is kept as-is.

Signed-off-by: mahphalke <Mahesh.Phalke@analog.com>

amiclaus commented on drivers/platform/maxim/max32670/maxim_gpio_irq.c:
```diff
@@ -0,0 +1,381 @@
+	return 0;
+
+error:
+	no_os_list_remove(actions);
```
this is not needed. the `free` happens already in `the no_os_list_init` if errors occur.

mphalke replied:
Make sense, will remove this check.

amiclaus commented on drivers/platform/maxim/max32670/maxim_irq.c:
```diff
@@ -0,0 +1,581 @@
+ */
+static bool is_gpio_irq_id(uint32_t irq_id)
+{
+	for (uint32_t i = 0; i < MXC_CFG_GPIO_INSTANCES; i++) {
```
nit: to allign better with the linux drivers code style, variables are declared at the start of the function. check the entire code for this occurrences.

mphalke replied:
Will move variable declaration outside loop

amiclaus commented on drivers/platform/maxim/max32670/maxim_irq.c:
```diff
@@ -0,0 +1,581 @@
+ * @param data2 - Key
+ * @return 0 if the two are equal, any other integer otherwise
+ */
+int32_t irq_action_cmp(void *data1, void *data2)
```
use `int` for the return type of the functions. check the entire code for these occurrences.

mphalke replied:
no_os_irq_platform_ops structure specifies return type as 'int32_t' for all functions. Do I need to change type of these function definitions to int, as compiler will through warnings?

amiclaus commented on drivers/platform/maxim/max32670/maxim_irq.c:
```diff
@@ -0,0 +1,581 @@
+		}
+
+		break;
+
```
nit: extra line before and after `break;` check the entire code for these occurrences.

mphalke replied:
Will remove extra space before and after break

amiclaus commented on drivers/platform/maxim/max32670/maxim_irq.h:
```diff
@@ -0,0 +1,79 @@
+#include "max32670.h"
+#include "no_os_irq.h"
+#include "uart.h"
+
```
add:

```
/******************************************************************************/
/*************************** Types Declarations *******************************/
/******************************************************************************/
```

amiclaus reviewed (commented):
some minor comments on my side

mphalke commented:
V1.0:
Review comment fixes:

- Moved variable declarations outside the loop statements.
- Added one line before break and removed one line after break statement.
- Removed redundant call to no_os_list_remove() function.
